infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/StreamsSnapshotReader.java

```diff
@@ -264,7 +263,10 @@ private boolean currentStreamHasNext() {
 
     @Override
     public void reset(long ts) {
-        streamsToSend = new PriorityQueue<>(streams);
+        config.syncWithInfoTable();
```

Comment:
I think you can have a method which updates the persisted table info from the registry table and returns the config.  no need to have 2 sync calls.

Reply:
Also, if there is a change, these syncs are not updating the persistent table.  The table only gets updated if a new stream is discovered during log entry sync